ArchiSteamFarm/Steam/Data/InventoryDescription.cs

```diff
@@ -0,0 +1,364 @@
+	public uint MarketFeeApp {
+		get => (uint) ProtobufBody.market_fee_app;
+		private init => ProtobufBody.market_fee_app = (int) value;
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("market_hash_name")]
+	public string MarketHashName {
+		get => ProtobufBody.market_hash_name;
+		private init => ProtobufBody.market_hash_name = value;
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("market_name")]
+	public string MarketName {
+		get => ProtobufBody.market_name;
+		private init => ProtobufBody.market_name = value;
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("name")]
+	public string Name {
+		get => ProtobufBody.name;
+		private init => ProtobufBody.name = value;
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("owner_actions")]
+	public ImmutableHashSet<ItemAction> OwnerActions {
+		get => ProtobufBody.owner_actions.Select(static action => new ItemAction(action.link, action.name)).ToImmutableHashSet();
+		private init {
+			ProtobufBody.owner_actions.Clear();
+
+			foreach (ItemAction action in value) {
+				ProtobufBody.owner_actions.Add(
+					new CEconItem_Action {
+						link = action.Link,
+						name = action.Name
+					}
+				);
+			}
+		}
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("type")]
+	public string TypeText {
+		get => ProtobufBody.type;
+		private init => ProtobufBody.type = value;
+	}
+
+	[JsonDisallowNull]
+	[JsonInclude]
+	[JsonPropertyName("tags")]
+	internal ImmutableHashSet<Tag> Tags {
+		get => ProtobufBody.tags.Select(static x => new Tag(x.category, x.internal_name, x.localized_category_name, x.localized_tag_name)).ToImmutableHashSet();
+		private init {
+			ProtobufBody.tags.Clear();
+
+			foreach (Tag tag in value) {
+				ProtobufBody.tags.Add(
+					new CEconItem_Tag {
+						appid = AppID,
+						category = tag.Identifier,
+						color = tag.Color,
+						internal_name = tag.Value,
+						localized_category_name = tag.LocalizedIdentifier,
+						localized_tag_name = tag.LocalizedValue
+					}
+				);
+			}
+		}
+	}
+
+	[JsonInclude]
+	[JsonPropertyName("tradable")]
+	[JsonRequired]
+	[JsonConverter(typeof(BooleanNumberConverter))]
+	internal bool Tradable {
+		get => ProtobufBody.tradable;
+		private init => ProtobufBody.tradable = value;
+	}
+
+	// For stubs and deserialization
+	[JsonConstructor]
+	internal InventoryDescription() { }
+
+	// Constructed from trades being received/sent
+	internal InventoryDescription(uint appID, ulong classID, ulong instanceID, bool marketable, bool tradable, IReadOnlyCollection<Tag>? tags = null) {
+		ArgumentOutOfRangeException.ThrowIfZero(appID);
+		ArgumentOutOfRangeException.ThrowIfZero(classID);
+
+		AppID = appID;
+		ClassID = classID;
+		InstanceID = instanceID;
+		Marketable = marketable;
+		Tradable = tradable;
+
+		if (tags?.Count > 0) {
+			Tags = tags.ToImmutableHashSet();
+		}
+	}
+
+	internal InventoryDescription(CEconItem_Description description) => ProtobufBody = description;
+
+	[UsedImplicitly]
+	public static bool ShouldSerializeAdditionalProperties() => false;
```

Comment:
We don't have any additional properties anymore IIRC, this can be removed therefore.